src/components/Accounts.tsx

```diff
@@ -38,19 +35,13 @@ const formatOptions = (accounts: Array<KeyringPair>) =>
   }));
 
 const Accounts = ({ className }: Props) => {
-  const accounts = useAccounts();
-  const derivedAccount = useDerivedAccount();
-  const { dispatchAccount } = useUpdateAccountContext();
+  const { account, accounts, derivedAccount, onChangeAccount } = useAccounts();
   const { sourceChain, targetChain } = useSourceTarget();
-  const { account: currentAccount } = useAccountContext();
 
-  const value = currentAccount?.address || '';
+  const value = account?.address || '';
 
   const onChange = (evt: any, { value }: any) => {
-    const account = accounts.find(({ address }) => address === value);
-    if (account) {
-      dispatchAccount({ payload: { account }, type: AccountActions.SET_ACCOUNT });
-    }
+    onChangeAccount(value);
```

Comment:
IMHO this should be called `setAccount` or `changeAccount`. `on*` prefix suggests that this is an event handler and not something that triggers the action.